dialer.go

```diff
@@ -97,6 +111,27 @@ func NewDialer(opts ...Option) (*Dialer, error) {
 		opt(c)
 	}
 
+	// if the chan of receivers == nil the checking will be skipped
+	for _, unreadyReceiver := range c.unreadyChs {
+		if unreadyReceiver == nil {
```

Comment:
return err if ch is nil, do not create channel because nobody would able to read from it

Reply:
changed to return error if channel == nil